libs/ts/decoders/src/encode-packed/utils/helpers.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+import type { ExpandedFieldOrArray } from '../../utils';
+
+import type { DecoderImplementations, DecoderLines } from './types';
+
+export const checkForDynamicData = (fields: ExpandedFieldOrArray[]) => {
+  let containsDynamicData = false;
+
+  fields.forEach(field => {
+    if (Array.isArray(field)) {
+      containsDynamicData = containsDynamicData || checkForDynamicData(field);
+    } else if (field.isDynamic) {
+      containsDynamicData = true;
+      return;
+    } else if ('components' in field) {
+      containsDynamicData =
+        containsDynamicData || checkForDynamicData(field.components!);
+    }
+  });
+
+  return containsDynamicData;
+};
+
+export const decoderImplementationsMap: Record<string, DecoderImplementations> =
+  {
+    cancun: {
+      generateDecoderPrimitiveLines: require('../helpers/primitiveFieldCancun')
```

Comment:
Make this a regular import at the top-level, as these files don't have any dependencies that we want to conditionally import